File: content-solutions/two-pointers/cover-window.js
function solution(street, recipe) {
  const need = Array(26).fill(0);
  for (const c of recipe) need[c.charCodeAt(0) - 97]++;
  let missing = recipe.length;
  let l = 0;
  let best = Infinity;
  for (let r = 0; r < street.length; r++) {
    const c = street.charCodeAt(r) - 97;
    if (need[c] > 0) missing--;
    need[c]--;
    while (missing === 0) {
      best = Math.min(best, r - l + 1);
      const left = street.charCodeAt(l) - 97;
      need[left]++;
      if (need[left] > 0) missing++;
      l++;
    }
  }
  return best === Infinity ? 0 : best;
}

File: content-solutions/two-pointers/cover-window.py
def solution(street, recipe):
    need = {}
    for c in recipe:
        need[c] = need.get(c, 0) + 1
    missing = len(recipe)
    l = 0
    best = float("inf")
    for r, c in enumerate(street):
        if need.get(c, 0) > 0:
            missing -= 1
        need[c] = need.get(c, 0) - 1
        while missing == 0:
            best = min(best, r - l + 1)
            left = street[l]
            need[left] += 1
            if need[left] > 0:
                missing += 1
            l += 1
    return 0 if best == float("inf") else best
